R/aggregate_kicking_stats.R

```diff
@@ -0,0 +1,114 @@
+      fg_made_50_59 = sum(as.numeric(.data$field_goal_result == "made") * dplyr::between(.data$kick_distance,50,59), na.rm = TRUE),
+      fg_made_60_ = sum(as.numeric(.data$field_goal_result == "made") * (.data$kick_distance >=60), na.rm = TRUE),
+      fg_missed_0_19 = sum(as.numeric(.data$field_goal_result == "missed") * dplyr::between(.data$kick_distance,0,19), na.rm = TRUE),
+      fg_missed_20_29 = sum(as.numeric(.data$field_goal_result == "missed") * dplyr::between(.data$kick_distance,20,29), na.rm = TRUE),
+      fg_missed_30_39 = sum(as.numeric(.data$field_goal_result == "missed") * dplyr::between(.data$kick_distance,30,39), na.rm = TRUE),
+      fg_missed_40_49 = sum(as.numeric(.data$field_goal_result == "missed") * dplyr::between(.data$kick_distance,40,49), na.rm = TRUE),
+      fg_missed_50_59 = sum(as.numeric(.data$field_goal_result == "missed") * dplyr::between(.data$kick_distance,50,59), na.rm = TRUE),
+      fg_missed_60_ = sum(as.numeric(.data$field_goal_result == "missed") * (.data$kick_distance >=60), na.rm = TRUE),
+      fg_made_list = .data$kick_distance[.data$field_goal_result == "made"] %>% na.omit() %>%  paste(collapse = ";"),
+      fg_missed_list = .data$kick_distance[.data$field_goal_result == "missed"] %>% na.omit() %>% paste(collapse = ";"),
+      fg_blocked_list = .data$kick_distance[.data$field_goal_result == "blocked"] %>% na.omit() %>% paste(collapse = ";")
+    ) %>%
+    dplyr::ungroup()
+
+  game_winners <- pbp %>%
+    dplyr::group_by(.data$game_id,.data$posteam) %>%
+    dplyr::filter(.data$fixed_drive == max(.data$fixed_drive)) %>%
+    dplyr::ungroup() %>%
+    dplyr::filter(.data$field_goal_attempt == 1, dplyr::between(.data$score_differential,-2,0)) %>%
+    dplyr::select(
+      "season",
+      "week",
+      "season_type",
+      "team" = "posteam",
+      "qtr",
+      "game_seconds_remaining",
+      "score_differential",
+      "desc",
+      "player_name" = "kicker_player_name",
+      "player_id" = "kicker_player_id",
+      "kick_distance",
+      "field_goal_attempt",
+      "extra_point_attempt",
+      "field_goal_result",
+      "extra_point_result"
+    ) %>%
+    dplyr::group_by(.data$season,.data$week,.data$season_type,.data$team,.data$player_name,.data$player_id) %>%
+    dplyr::summarise(
+      gwfg_att = n(),
+      gwfg_distance = .data$kick_distance,
+      gwfg_made = sum(.data$field_goal_result == "made", na.rm = TRUE),
+      gwfg_missed = sum(.data$field_goal_result == "missed", na.rm = TRUE),
+      gwfg_blocked = sum(.data$field_goal_result == "blocked", na.rm = TRUE),
+    ) %>%
+    dplyr::ungroup()
+
+  full_kicks <- base_kicks %>%
+    dplyr::left_join(
+      game_winners,
+      by = c("season", "week", "season_type", "team", "player_name", "player_id")) %>%
+    dplyr::mutate(
+      dplyr::across(dplyr::starts_with("gwfg"),tidyr::replace_na,0)
```

Comment:
We have avoided using dplyr >= 1.0 functions to be compatible with older versions. I think I would stick to that for one more season so we can’t use stuff like `across` or `relocate`